docs/courses/robotics/3.5-computer-vision.md

```diff
@@ -0,0 +1,157 @@
+- [AprilTag Python Library](https://pypi.org/project/apriltag/)
+- [Motorized Microscope]
+- [Raspberry Pi Camera Module](https://www.raspberrypi.org/products/camera-module-v2/)
+- [USB-A to micro USB-B cable]
+
+#### Demo
+
+✅ Read the [OpenCV Documentation](https://docs.opencv.org/4.x/)
+
+✅ Watch [Introduction to OpenCV](https://www.youtube.com/watch?v=oXlwWbU8l2o)
+
+In this task, you will use **OpenCV** to perform image preprocessing, such as applying a Gaussian blur, detecting edges, and identifying regions of interest (ROI).
+
+```python
+import cv2
+import numpy as np
+
+# Load an image
+img = cv2.imread('sample_image.jpg')
+
+# Apply Gaussian Blur
+blurred = cv2.GaussianBlur(img, (5, 5), 0)
+
+# Edge Detection
+edges = cv2.Canny(blurred, 100, 200)
+
+cv2.imshow('Edges', edges)
+cv2.waitKey(0)
+cv2.destroyAllWindows()
+```
+
+The above code will help you preprocess the image to smoothen it and then detect edges. This is the first step before applying any region of interest search.
+
+---
+
+### AprilTags
+
+Next, you will utilize **AprilTags**, a popular fiducial marker system for reliable and robust spatial referencing and object identification. AprilTags are often used in robotics for locating objects and navigating environments.
+
+#### Bill of Materials
+
+- [AprilTags Python Library](https://pypi.org/project/apriltag/)
+- [AprilTags Documentation](https://april.eecs.umich.edu/software/apriltag.html)
+- [Raspberry Pi Camera Module](https://www.raspberrypi.org/products/camera-module-v2/)
+- [Motorized Microscope]
+#### Demo
+
+✅ Watch [Vision Programming with AprilTags](https://youtu.be/TG9KAa2EGzQ)
+
+The following code demonstrates how to use **AprilTags** for spatial referencing by detecting tags and performing ID lookup.
+
+```python
+import apriltag
+import cv2
+
+# Load an image with AprilTags
+img = cv2.imread('apriltag_image.jpg')
+gray_image = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+
+# Initialize AprilTag detector
+detector = apriltag.Detector()
+
+# Detect tags
+tags = detector.detect(gray_image)
+
+# Display tag information
+for tag in tags:
+    print(f"Detected tag ID: {tag.tag_id}")
+    cv2.rectangle(img, 
+                  (int(tag.corners[0][0]), int(tag.corners[0][1])), 
+                  (int(tag.corners[2][0]), int(tag.corners[2][1])), 
+                  (0, 255, 0), 2)
+
+cv2.imshow('AprilTag Detected', img)
+cv2.waitKey(0)
+cv2.destroyAllWindows()
+```
+
+In this demo, you will detect **AprilTags** in an image and draw bounding boxes around them. You can then use the tag ID for further referencing tasks.
+
+---
+
+### Motorized Microscope and OpenCV
+
+After detecting regions of interest (ROI) using **OpenCV** and **AprilTags**, you will integrate a motorized microscope. The microscope will move automatically based on the ROI found in the image, providing fine control and automation for analyzing samples.
+
+#### Demo
+
+✅ Watch [Controlling a Motorized Microscope](https://youtu.be/fHZSsAKThW4)
```

Comment:
Replace with content related to OpenFlexure Microscope v7, if possible

https://openflexure.org/